peer/bind.go

```diff
@@ -31,91 +31,95 @@ import (
 	"go.uber.org/multierr"
 )
 
-// Binder is a callback, provided to peer.Bind, that accepts a peer list and
-// binds it to a peer provider for the duration of the returned lifecycle.
-// The lifecycle that the binder returns should start and stop binding peers to
-// the list.
-// The binder must not call block on updating the list, because that will
-// typically block until the peer list has started.
-type Binder func(peer.List) transport.Lifecycle
-
-// Bind couples a peer list with a peer provider.
-// The peer provider must produce a binder that takes the peer list and returns
-// a lifecycle for the duration of the binding.
-func Bind(chooserList peer.ChooserList, bind Binder) *BoundChooser {
+// Bind couples a peer list with a peer list updater.
+// Bind accepts a peer list and passes that peer list to a binder.
+// The binder must return a peer list updater bound to the peer list.
+// The peer list updater must implement Lifecycle so the bound peer list
+// can start and stop updates.
+func Bind(chooserList peer.ChooserList, bind peer.Binder) *BoundChooser {
 	return &BoundChooser{
-		once:    intsync.Once(),
-		binding: bind(chooserList),
-		chooser: chooserList,
+		once:        intsync.Once(),
+		chooserList: chooserList,
+		updater:     bind(chooserList),
 	}
 }
 
-// BoundChooser is a peer chooser that couples a peer list and a peer provider
-// for the duration of its lifecycle.
+// BoundChooser is a peer chooser that couples a peer list and a peer list
+// updater for the duration of its lifecycle.
 type BoundChooser struct {
-	once    intsync.LifecycleOnce // the lifecycle of the bound peer chooser+provider
-	binding transport.Lifecycle   // the lifecycle of the peer provider
-	chooser peer.Chooser          // the peer chooser and the lifecycle of the peer chooser
+	once        intsync.LifecycleOnce
+	updater     transport.Lifecycle // the peer list updater, which to us is just a lifecycle
+	chooserList peer.ChooserList    // the peer list/chooser, also a lifecycle
```

Comment:
That is a detail I don’t want to depend on someone to incidentally recall here, since the BoundChooser is responsible for lifecycle management.